### DISCUSSION TITLE
generate both entities and migration at the same time would be a useful cli function

### DISCUSSION THREAD

**Discussion by AshWang-whm** [SeaQL/sea-orm#2028] [Ideas]
I build a project in db first , I can generate entities with cli , but I can't generate the migration with db first, and then I have to export the db file and package into project folder, it would be useful that generate entities and migrations at the same time.

**Comment by fu050409** [comment]
I think we can allow migration to be generated based on Entity classes. In that case, we would only need to design the entity classes and then generate and migrate the database based on the entity classes. At the same time, the corresponding migration files can be generated based on the changes to the entity classes.

**Reply by AshWang-whm** [reply]
migration generated base on Entity classes may be more difficult than generated from database, Entity classes do not have the database field type or the others infos.

**Reply by fu050409** [reply]
You are right, it is indeed harder to achieve. But it is undeniable that it is more convenient. Of course, this is just my simple idea, and the specific solution will of course depend on the maintainer.

All I know is that it is easy to implement in Python, but may be far more complicated in Rust.
